src/modules/auth/stores/auth-store.ts

```diff
@@ -0,0 +1,25 @@
+import {create} from 'zustand';
+import {User} from '../../../models/user';
+import {setIsAuthenticatedStack} from '../../../navigations/utils/setIsAuthenticatedStack';
+
+type AuthStore = {
+  user: User | null;
+  token: string | null;
+};
+
+export const INITIAL_STORE = {
+  token: null,
+  user: null,
+};
+
+const useStore = create<AuthStore>(() => INITIAL_STORE);
+
+useStore.subscribe(({token}) => {
+  const isAuthenticated = !!token;
+  setIsAuthenticatedStack(isAuthenticated);
+});
+
+export const Auth = {
+  useStore,
+  isAuthenticated: !!useStore.getState().token,
+};
```

Comment:
Eu to achando que vc não precisa dessa store. Me parece que lá no arquivo `use-auth` vc apenas seta estado, e aqui vc usa o estado pra obter a informação `isAuthenticated`. Porém vc poderia obter essa informação no início da execução do app consultando o local-storage, numa lógica de useEffect.

Reply:
nao to conseguindo fazer, vou implementar a parte da infinity scroll e depois faço uma branch de fix pra resolver se nao vou acabar demorando muito nessa track